main.go

```diff
@@ -19,18 +19,49 @@ package main
 import (
 	"flag"
 	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
 	"github.com/golang/glog"
 
 	"k8s.io/metacontroller/apis/metacontroller/v1alpha1"
 	internalclient "k8s.io/metacontroller/client/generated/clientset/internalclientset"
+	internalinformers "k8s.io/metacontroller/client/generated/informer/externalversions"
+	internallisters "k8s.io/metacontroller/client/generated/lister/metacontroller/v1alpha1"
 
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 )
 
-func resyncAll(config *rest.Config) error {
+// metaController manages metacontroller API objects.
+type metaController struct {
```

Comment:
This seems to hold the same place semantically as our `SharedInformerFactory`, just with aliases for some long "getter" calls. Were you thinking there would be additional logic here? If not, I'd prefer to just pass around the `SharedInformerFactory` directly, so we don't have to keep this struct in sync with it.

Reply:
That's reasonable. I usually use this pattern where structs contain other references like workqueues, "additional logic" to quote your words. No real need for this here.